### ISSUE TITLE
Word and paragraph selection doesn't work for `TextInput` placed inside `Flickable`

### ISSUE THREAD

**Issue by lmaxyz** [slint-ui/slint#6514]
Word and paragraph selection by double and triple clicks doen't work for `TextInput` placed inside `Flickable`.

Simple example:
```slint
Flickable {
    width: parent.width;
    viewport-width: txt.preferred-width;
    txt:=TextInput {
        text: "Some text to select it and check TextInput component. Try to select with double and triple click.";
    }
}
```
I figured out it happens because of these lines of code:
https://github.com/slint-ui/slint/blob/ccf5f04087c5d920c80e7e706230e99a12b9e9ef/internal/core/input.rs#L827-L829

Why does it exist? I didn't understand.
If I commented this code, double and triple clicks becomes to work properly - words and paragraphs selected as expected. And in my case there is no side effects.

**Comment by ogoffart** [comment]
Thanks for filing this bug report.
I can confirm the bug.

Git blame point to this commit from me: https://github.com/slint-ui/slint/commit/808db750221d84ffcf48deae213ad981b09e3094
The goal of this test is to make sure that that the double-click doesn't trigger if the click was done first on one element, and then on an other element (like clicking fast on one element; then on another real close (because of the 10px threshold arleady applied)
We could therefore get a double click without getting a single click before

I think we probably do need this test.
But maybe the test is wrong or the last_top_item is wrong. When there is a delayed item we may need to use the delayed item instead.

**Comment by lmaxyz** [comment]
> When there is a delayed item we may need to use the delayed item instead.

If you are talking about items that return `DelayForwarding` event result, I tried to replace it to just `Forwarding` or another types of results - it was changing behavior but didn't fix zero click count. Maybe I did something wrong.

And may be it's important, `set_click_count(0)` invokes as times as count of items placed inside `Flickable`.
Another one thing - maybe the problem is what `TextInput` returns when handles events. I tried to change returned event results, but it didn't work.

**Comment by ogoffart** [comment]
Fixed in https://github.com/slint-ui/slint/pull/11610
